app/src/lib/stores/app-store.ts

```diff
@@ -2585,16 +2600,14 @@ export class AppStore extends TypedBaseStore<IAppState> {
       return
     }
 
-    if (tryBackgroundFetch) {
-      const lastPush = await inferLastPushForRepository(
-        this.accounts,
-        gitStore,
-        repository
-      )
+    const lastPush = await inferLastPushForRepository(
+      this.accounts,
+      gitStore,
+      repository
+    )
 
-      if (this.shouldBackgroundFetch(repository, lastPush)) {
-        await this._fetch(repository, FetchType.BackgroundTask)
-      }
+    if (this.shouldBackgroundFetch(repository, lastPush)) {
+      await this._fetch(repository, FetchType.BackgroundTask)
     }
 
     lookup.set(repository.id, {
```

Comment:
i wish there was a clean way to use `updateSidebarIndicator` here without doing extra work since the logic is so similar but i can't quite figure out a way to do that right now

Reply:
Yeah, I stared at this a bunch while I was in here and couldn't really tease out something I liked either. 

Gonna keep thinking about it, but I won't hold up this PR and the benefits it already gives.

Reply:
> Gonna keep thinking about it, but I won't hold up this PR and the benefits it already gives.

yep, totally agree